src/controllers/router-store/utils/index.ts

```diff
@@ -64,3 +69,27 @@ export const updateQueryParams = (location: Location, query: Query): string => {
 export const getRelativeURLFromLocation = (location: Location): string => {
   return `${location.pathname}${location.search}${location.hash}`;
 };
+
+export const shouldReload = ({
+  context,
+  prevContext,
+  pluginId,
+  defaultShouldReload,
+}: Parameters<ShouldReloadFunction>[0]) => {
+  if (
+    context.route === prevContext.route &&
+    context.route.EXPERIMENTAL__shouldReload
+  ) {
+    const routeChoice = context.route.EXPERIMENTAL__shouldReload({
+      context,
+      prevContext,
+      pluginId,
+      defaultShouldReload,
```

Comment:
Why would a route need to care about the default value? It can be inferred by the `pluginId` as well, if needed